pyroomacoustics/bss/sparseauxiva.py

```diff
@@ -0,0 +1,161 @@
+"""
```

Comment:
We are now adding copyright and license information in a special header to the files. You can follow the following pattern.
```
# <one liner describing file content>
# Copyright (C) 2019  <names go here>
#
# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice shall be included in all
# copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
# SOFTWARE.
#
# You should have received a copy of the MIT License along with this program. If
# not, see <https://opensource.org/licenses/MIT>.
r"""
The rest of the docstring goes here.

Examples:

more stuff...

etc
"""
```
The reason is that as an open source project, we should have the license information at the head of every file (slowly working to get there), but we don't want to have this information on every page of the documentation. The comment using `#` is ignored by the documentation so that only the part in triple quotes `"""` is used for docstrings.

Reply:
What if a file has functions made by different authors (such as common.py in my pull request)? Should I (1) pile all the authors in one license information header, (2) copy the license information to each function with its corresponding authors or (3) copy only one license information header and mention the authors in the docstring of each function?

Reply:
I think quite often the name of the authors are all on the copyright line. We could also add more in details all the authors in a dedicated file tracking contributions (that file doesn't exist yet).